app/controllers/constraints_controller.rb

```diff
@@ -2,36 +2,36 @@
 
 class ConstraintsController < AuthenticationController
   before_action :set_planning_application
-  before_action :ensure_constraint_edits_unlocked, only: %i[edit update]
+  before_action :ensure_constraint_edits_unlocked, only: %i[show update]
   before_action :set_planning_application_constraints, only: %i[update]
-  before_action :set_removed_constraints, only: %i[update]
 
   def show; end
 
-  def edit
-    @constraints = @planning_application.constraints
-    planning_application_constraints = @planning_application.planning_application_constraints.includes(:constraint)
-    @constraint_ids = planning_application_constraints.pluck(:constraint_id)
-  end
-
   def update
     ActiveRecord::Base.transaction do
-      constraint_ids.each do |constraint_id|
-        @planning_application_constraints.find_or_create_by!(constraint_id:)
+      @planning_application_constraints.each do |pac|
```

Comment:
Tiny nitpick but is using `constraint` rather than `pac` a bit more readable here

Reply:
It's named as `pac`, as is an object of `planning_application_constraints` and not `constraints`